src/coreclr/vm/jitinterface.cpp

```diff
@@ -4671,6 +4670,49 @@ unsigned CEEInfo::getArrayRank(CORINFO_CLASS_HANDLE  cls)
     return result;
 }
 
+/*********************************************************************/
+// Get the index of runtime provided array method
+CorInfoArrayIntrinsic CEEInfo::getArrayFuncIndex(CORINFO_METHOD_HANDLE ftn)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    unsigned result = CorInfoArrayIntrinsic::ILLEGAL;
+
+    JIT_TO_EE_TRANSITION();
+
+    MethodDesc* pMD = GetMethod(ftn);
+
+    if (pMD->IsArray())
+    {
+        TypeHandle th = TypeHandle(pMD->GetMethodTable());
+
+        if (th.GetInternalCorElementType() != ELEMENT_TYPE_SZARRAY)
+        {
```

Comment:
```suggestion
```
This should not be needed.

Reply:
I adapted this logic from `CEEInfo::isSDArray` implementation above. Should we simplify it as well?

```diff
-        result = (th.GetInternalCorElementType() == ELEMENT_TYPE_SZARRAY);
+        result = true;
```

Reply:
`isSDArray` should stay as is.

The array intrinsic methods exist on SD arrays as well. The C# compiler won't use them on SD arrays, but they may be used by other compilers. The JIT should be able to see them and decide whether it wants to do anything about them. It is what the existing code does at least.